### PR TITLE
Optimized network calls

### PR DESCRIPTION
**Pull request by mickaelandrieu**
This contributions is not that huge!

The first commit puts all the badge images from gamification server into the module (and updates the path on badge image url).

The second commit optimizes and fix the function `isFresh` responsible of calling the server if files are outdated: as the function was broken, the CSS files from advices were requested everytime!

> This doesn't solve the resiliency issue for now. I'm waiting for https://github.com/PrestaShop/gamification/pull/24 to be merged as I want to write integration tests to confirm the module is totally functional even when gamification server is down.

### CODE REVIEW

File: gamification.php [hunk 1]
```diff
@@ -119,10 +121,11 @@ public function installTab()
         if (version_compare(_PS_VERSION_, '1.7.0.0', '>=')) {
             //AdminPreferences
             $tab->id_parent = (int)Db::getInstance(_PS_USE_SQL_SLAVE_)
-                                ->getValue('SELECT MIN(id_tab)
-											FROM `'._DB_PREFIX_.'tab`
-											WHERE `class_name` = "'.pSQL('ShopParameters').'"'
-                                        );
+                ->getValue('SELECT MIN(id_tab)
+				                 FROM `'._DB_PREFIX_.'tab`
```
Comment (PierreRambaud) [inline gamification.php]: weird indent.

File: gamification.php [hunk 2]
```diff
@@ -547,13 +552,13 @@ public function getFormatedConditionsIds()
...
-            if (filesize($file) < 1) {
-                return false;
-            }
-            return ((time() - @filemtime($file)) < $timeout);
-        } else {
+        if (!file_exists($file)) {
             return false;
         }
+
+        $lastFileUpdate = date("F d Y H:i:s.", filemtime($file) + $timeout);
```
Comment (PierreRambaud) [inline gamification.php]: Simple quote

File: gamification.php [hunk 3]
```diff
@@ -547,13 +552,13 @@ public function getFormatedConditionsIds()
...
+        $now = date("F d Y H:i:s.");
+
+        return $now < $lastFileUpdate;
```
Comment (PierreRambaud) [inline gamification.php]: Date string are not really compared here. Use strtotime, Datetime, or directly timestamp.

File: gamification.php (hunk 1 above)
Comment (mickaelandrieu) [reply gamification.php]: we could use PHP CS Fixer but ... oh wait! waiting for => #24

File: gamification.php (hunk 2 above)
Comment (mickaelandrieu) [reply gamification.php]: we could use PHP CS Fixer but ... oh wait! waiting for => #24

File: gamification.php (hunk 3 above)
Comment (mickaelandrieu) [reply gamification.php]: It is intended, I think it's more readable here, and easier to debug in case of problem.

Comment (PierreRambaud) [reply gamification.php]: You can compare date with the string format. it's just not working :/

```php
$now = date("F d Y H:i:s.", 1539876652);
$time = date("F d Y H:i:s.", 1545150546);

echo $now . PHP_EOL;
echo $time . PHP_EOL;
var_dump($now < $time);
```

```bash
[3.6.3] got@debian:~$php test.php 
October 18 2018 17:30:52.
December 18 2018 17:29:06.
bool(false)
```

The real test needs to be 
```php
$lastFileUpdate = filemtime($file) + $timeout;
$now = time();
var_dump($now < $lastFileUpdate);
```

**Pull request merged**